core/camel-xml-io/src/main/java/org/apache/camel/xml/io/MXParser.java

```diff
@@ -0,0 +1,3306 @@
+        }
+        boolean gotS = false;
+        final boolean normalizeIgnorableWS = tokenize == true && roundtripSupported == false;
+        boolean normalizedCR = false;
+        try {
+            // epilog: Misc*
+            char ch;
+            if(seenMarkup) {
+                ch = buf[ pos - 1 ];
+            } else {
+                ch = more();
+            }
+            seenMarkup = false;
+            posStart = pos - 1;
+            if(!reachedEnd) {
+                while(true) {
+                    // deal with Misc
+                    // [27] Misc ::= Comment | PI | S
+                    if(ch == '<') {
+                        if(gotS && tokenize) {
+                            posEnd = pos - 1;
+                            seenMarkup = true;
+                            return eventType = IGNORABLE_WHITESPACE;
+                        }
+                        ch = more();
+                        if(reachedEnd) {
+                            break;
+                        }
+                        if(ch == '?') {
+                            // check if it is 'xml'
+                            // deal with XMLDecl
+                            parsePI();
+                            if(tokenize) return eventType = PROCESSING_INSTRUCTION;
+
+                        } else if(ch == '!') {
+                            ch = more();
+                            if(reachedEnd) {
+                                break;
+                            }
+                            if(ch == 'D') {
+                                parseDocdecl(); //FIXME
+                                if(tokenize) return eventType = DOCDECL;
+                            } else if(ch == '-') {
+                                parseComment();
+                                if(tokenize) return eventType = COMMENT;
+                            } else {
+                                throw new XmlPullParserException(
+                                        "unexpected markup <!"+printable(ch), this, null);
+                            }
+                        } else if(ch == '/') {
+                            throw new XmlPullParserException(
+                                    "end tag not allowed in epilog but got "+printable(ch), this, null);
+                        } else if(isNameStartChar(ch)) {
+                            throw new XmlPullParserException(
+                                    "start tag not allowed in epilog but got "+printable(ch), this, null);
+                        } else {
+                            throw new XmlPullParserException(
+                                    "in epilog expected ignorable content and not "+printable(ch),
+                                    this, null);
+                        }
+                    } else if(isS(ch)) {
+                        gotS = true;
+                        if(normalizeIgnorableWS) {
+                            if(ch == 
```

Comment:
This was reported as logical alert [here](https://lgtm.com/projects/g/apache/camel/snapshot/3648e893d7d64ed125579faffb3f95f45166a028/files/core/camel-xml-io/src/main/java/org/apache/camel/xml/io/MXParser.java?sort=name&dir=ASC&mode=heatmap#L1928). This is because of the `xmlnsPos < 5`. Is it possible to address that?